dataloader: create new tables and populate with sample_data from csv

atla5 opened the pull request:
Resolves #7 

### Description of changes
- [x] manually create new tables via H2 console 
- [x] create new `DataLoader` with main method for loading data 
- [x] add `insertNew*` helpers to each `*Proxy` class.
- [x] use the existing controller connection to update the tables through the code (instead of manually)
- [x] update `vendors` table to add in the vendor name.
- [ ] ~remove all `date`s and customer demographics from db~ (moved to a separate PR)

_note: table creation itself was done manually in H2, and will need to be updated once schema and sample data are all confirmed with the team_

atla5 commented on src/main/java/com/rainforestcommerce/rcdb/controllers/DataLoader.java:
```diff
@@ -0,0 +1,169 @@
+public class DataLoader {
+    public static String GENERIC_INSERT_STATEMENT = "INSERT into %s VALUES %s;";
+
+    private static String dataDirectory = "/Users/aidan/projects/school/csci-320/src/main/resources/sample_data";
```
need to make this generic. 👍

atla5 commented on src/main/java/com/rainforestcommerce/rcdb/controllers/DataLoader.java:
```diff
@@ -0,0 +1,169 @@
+
+    public static boolean insertValuesIntoTable(String values, String tableName){
+        String insertCommand = String.format(GENERIC_INSERT_STATEMENT, tableName, values);
+        System.out.println(insertCommand);
```
switch to `LOGGER` and set to `debug`

grahamhome commented:
Resolved conflicts with master. @atla5 I cannot approve this PR yet as I cannot run it - where is the code to create the tables?

atla5 committed:
dataloader: add new column for 'vendor_name'

atla5 committed:
dataloader: [enhancement] hide loaders behind static boolean

atla5 committed:
config: update .gitignore to reduce commit noise

atla5 committed:
dataloader: rm test.trace.db (just error logs)

h2 puts all the error logs from run commands in there, and it's not really useful for our purposes

atla5 commented:
the `create_tables.sql` ticket is being created in #18, so it should be approved without it.

(i'd answered your question quite a while ago on slack, but figured i should add it here too for provenance)

atla5 committed:
Merge branch 'master' of https://github.com/atla5/csci320_rc into dataloader

aaa1008 reviewed (approved):
Looks good to me.
